Fix FormRequest.formdata with GET method duplicates same key in querystring (#2919)

iamriel opened the pull request:
This is the fix for issue #2919 This is my first time contributing to an open source project.  This PR is based on gekco's PR [here](https://github.com/scrapy/scrapy/pull/3151).  I followed the instruction in [Submitting Patches](https://doc.scrapy.org/en/master/contributing.html#submitting-patches) for updating an existing PR.  Hopefully I have done it correctly.

HassanQamar07 commented on scrapy/http/request/form.py:
```diff
@@ -33,7 +33,11 @@ def __init__(self, *args, **kwargs):
-                self._set_url(self.url + ('&' if '?' in self.url else '?') + querystr)
+                query = dict(parse_qsl(urlparse(self.url).query))
+                if query:
+                    query.update(dict(items))
```
Sometimes we've to send multiple values for a single key, `dict(items)` will make it impossible and will result in unexpected URL.

`
formdata=[
('a','1'),
('a','2'),
('b','3'),
]
`

HassanQamar07 commented:
@iamriel Can you add another test case which tests the multiple values for same key name
e.g

`formdata=[('a','a')),('a','A')),('b','B')),]`

iamriel commented:
@HassanQamar07 Thanks for checking this PR, you are right about it, I have updated the implementation and added the test in this [commit](https://github.com/iamriel/scrapy/commit/db7b4250).  Please let me know if there is anything else. Thanks!

iamriel commented:
Ooop, the build failed, I'll update later today.

maramsumanth commented:
@iamriel , what happens when there is fragment in url. I think your code fails.

iamriel commented:
I apologize I haven't had time to attend into this.  I'm closing this as @maramsumanth already had an approved PR for this. Thanks everyone!
